sched/pthread/pthread_create.c

```diff
@@ -529,37 +521,28 @@ int nx_pthread_create(pthread_trampoline_t trampoline, FAR pthread_t *thread,
   pid = ptcb->cmn.pid;
   pjoin->thread = (pthread_t)pid;
 
-  /* Initialize the semaphores in the join structure to zero. */
+  /* Initialize the semaphore in the join structure to zero. */
 
-  ret = nxsem_init(&pjoin->data_sem, 0, 0);
```

Comment:
Why initialization of `pjoin->data_sem` is removed from here?

Reply:
don't need it,data_sem is used to make sure a thread start when create，in this pacth,delete it,The new thread created will be handed over to the scheduler to start.

Reply:
> Why initialization of `pjoin->data_sem` is removed from here?

@pkarashchenko since it isn't used here, instead pthread_notifywaiters will do the initialization:
https://github.com/apache/incubator-nuttx/blob/master/sched/pthread/pthread_completejoin.c#L68

Reply:
Yes. I revisit changes again and see that it is not needed anymore. But due to this I've found that `nxsem_set_protocol(&pjoin->data_sem, SEM_PRIO_NONE);` is missing after `nxsem_init(&pjoin->data_sem, 0, (ntasks_waiting + 1));` in `pthread_notifywaiters` :)

Reply:
@anjiahao1 please create a new patch to fix the issue mentioned by @pkarashchenko .